eng/cmake/global_compile_options.txt

```diff
@@ -5,6 +5,12 @@ if(MSVC)
   endif()
 
   add_compile_options(/W4 ${WARNINGS_AS_ERRORS_FLAG} /wd5031 /wd4668 /wd4820 /wd4255 /wd4710 /analyze)
+
+  add_compile_options(
+    $<$<CONFIG:>:/MT> #---------|
+    $<$<CONFIG:Debug>:/MTd> #---|-- Statically link the runtime libraries
+    $<$<CONFIG:Release>:/MT> #--|
+  )
```

Comment:
@barcharcraz and @BillyONeal - can you please help review and sign-off on this approach.

Reply:
Just for reference I pulled it from here:
https://gitlab.kitware.com/cmake/cmake/-/issues/18390